File: 115_tap_quan_su.py
def sinhviennghi():
    global cauhinhcuoi  , n , k , M , A 
    if cauhinhcuoi == True:
        return k 
    nghi= k
    for i in range (0, k):
        for j in range(0 , k):
            if M[i]==A[j]: 
                nghi -=1
    return nghi
def sinh():
    global cauhinhcuoi ,n , k , M
    i = k-1 
    while M[i]==n-k+i+1 and i>=0:
        i-=1
    if i==-1:
        cauhinhcuoi = True
        return 
    M[i]+=1
    for j in range(i+1 ,k):
        M[j] = M[i] + j- i
t = int(input())
while t>0:
    n , k = map(int , input().split())
    cauhinhcuoi  = False
    M= list(map(int , input().split()))
    A = []
    for i in range( 0, k ):
        A.append(M[i])
    sinh()
    print(sinhviennghi())
    t-=1 

File: 115_tap_quan_su.cpp
#include<iostream>
using namespace std ; 
int n ,  k , M[40] , A[40] ; 
bool cauhinhcuoi ;
int sinhviennghi() {
    if (cauhinhcuoi == true ) {
        return k ; 
    } else {
    int nghi = k ; 
    for(int i=1 ; i<=k ; i++) {
        for(int j=1 ; j<=k ; j++) {
            if( M[i] == A[j] ) nghi-- ; 
        }
    }
    return nghi ;       
    }
}
void sinh() {
    int i =  k ; 
    while(M[i] == n- k+i and i>=1 ) i-- ; 
    if( i==0 ) {
        cauhinhcuoi = true; return ; 
    }
    M[i] += 1 ; 
    for(int j=i ; j<=k  ; j++) {
        M[j] = M[i]+ j-i ; 
    }
}
main() {
     int t; cin>>t ; 
     while(t--) {
         cin>>n>>k ;
         cauhinhcuoi = false ;  
         for(int i=1 ; i<=k ;i++) {
             cin>>M[i] ; 
             A[i] = M[i] ; 
         }
        sinh(); 
        cout<<sinhviennghi()<<endl ; 
     }
}